### PR TITLE
ncm-network: add support for device configuration dropin file

### PR DESCRIPTION
**Pull request by aka7**
provide support to create drop-in file for device config and add option to add keep-configuration option

fixes [1711](https://github.com/quattor/configuration-modules-core/issues/1711)

* Why the change is necessary.
To able to turn off device keep-configuration option.
* What backwards incompatibility it may introduce.
None.

### CODE REVIEW

**Comment by aka7** [comment]
The schema update  is only for nmstate backend, but I'm not totally sure how to include this in nmstate schema, from the separation @stdweird did.  So some input would be appreciated.

File: ncm-network/src/main/perl/nmstate.pm [hunk 1]
```diff
@@ -103,6 +104,26 @@ sub disable_nm_manage_dns
...
+    my @data = ('[device]');
+
+    if ( $nm_device_config ) {
+        foreach my $key (sort keys %$nm_device_config){
+            push @data, $key."=".$nm_device_config->{$key};
+        };
+
+        $self->verbose("setting device configuration dropin");
+    }
+    my $fh = CAF::FileWriter->new($NM_DEVICE_DROPIN_CFG_FILE, mode => oct(444), log => $self, keeps_state => 1);
```
Comment (ned21) [inline ncm-network/src/main/perl/nmstate.pm:119]: keeps_state seems wrong here if a successful write results in a network reload?

Comment (aka7) [reply ncm-network/src/main/perl/nmstate.pm:119]: I don't think I fully understand what keeps_state purpose is?

Comment (jrha) [reply ncm-network/src/main/perl/nmstate.pm:119]: It means that writing the file doesn't affect the state of the system, which isn't the case here?

> **keeps_state**
>
> A boolean specifying whether a file change respects the current system state or not. A file with keeps_state will be created/modified, regardless of any value for NoAction. This is useful when creating temporary files that are required for a NoAction run.
> By default, file changes modify the state and thus keeps_state is false.

https://quattor-documentation.readthedocs.io/en/stable/CAF/CAF_FileWriter.html#public-methods

Comment (aka7) [reply ncm-network/src/main/perl/nmstate.pm]: ah thanks, will remove that then if default is false, also may need to fix it in sub disable_nm_manage_dns() too.

**Comment by stdweird** [comment]
@aka7 not sure the comment https://github.com/quattor/configuration-modules-core/pull/1659#issuecomment-2361418243 was supposed to be added here ?

**Comment by aka7** [comment]
Probably.  once the schema separation is understood and works, happy to
move it under nmstate schema only. I needed this quick to fix a bug we
noticed with network manager. So happy to move it under nmstate schema, if
that's what you mean?

**Comment by jrha** [comment]
Schema migration in #1720.

**Comment by jrha** [comment]
Rebased and updated to use nmstate specific schema.

**Review by jrha — APPROVED** [review]

**Pull request merged**